pandas/tests/series/test_reductions.py

```diff
@@ -9,6 +9,27 @@
 import pandas._testing as tm
 
 
+@pytest.mark.parametrize("as_period", [True, False])
+def test_mode_extension_dtype(as_period):
+    # GH#41927 preserve dt64tz dtype
+    ser = Series([pd.Timestamp(1979, 4, n) for n in range(1, 5)])
+
+    if as_period:
+        ser = ser.dt.to_period("D")
+    else:
+        ser = ser.dt.tz_localize("US/Central")
+
+    res = ser.mode()
+    assert res.dtype == ser.dtype
+    tm.assert_series_equal(res, ser)
+
+    res = pd.core.algorithms.mode(ser._values)
```

Comment:
don't we have a rule which prevents this? e.g. can you import at the top instead

Reply:
not sure about the rule.  will move the import